Fix USB problems on the LPC1768

c1728p9 opened the pull request:
### Description

Implement the missing abort function and fix an error caused by not resetting data toggle.

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

c1728p9 committed:
Fix LPC1768 test failures

Reset endpoints when they are removed. This ensures buffers and the
data toggle bit get reset back to zero.

c1728p9 committed:
Implement LPC1768 USB abort function

Implement the USB abort function so interrupts wont fire for a
transfer that has been aborted. The transfer may still take place
but the buffer passed into endpoint_write or endpoint_read will not
be used.

c1728p9 commented:
@fkjagodzinski @maciejbocianski @mmahadevan108

c1728p9 commented:
/morph test

fkjagodzinski reviewed (commented):
I've checked this PR against my updated tests from #6880 and I think there is something missing when reenabling endpoint interrups. When `endpoint_abort()` is called, and then a new interface(or altsetting) is set, the IN/OUT callbacks are not called anymore.

c1728p9 commented:
@fkjagodzinski I updated this PR and it should resolve the problems with #6880. I tracked down the cause of the failure to disabling the endpoint event in USBEpIntEn. It had some nasty side effects causing the failure. I removed use of this and added a warning in the code.

fkjagodzinski reviewed (approved):
LGTM!

cmonr commented:
/morph build

cmonr commented:
@c1728p9 Feel free to check out the build failure, but a quick look is pointing to this being an example issue, not a branch/PR issue.

cmonr commented:
/morph build

cmonr commented:
/morph export-build

cmonr commented:
Had to stop and restart export build due to momentary IAR license issues.

cmonr commented:
Halting CI builds until RC3 PRs are completed. Will resume after.

0xc0170 commented:
/morph build
